File: src/main/java/io/github/eone666/telegramnotifier/utils/telegram/Telegram.java
package io.github.eone666.telegramnotifier.utils.telegram;

import io.github.eone666.telegramnotifier.utils.Request;
import org.bukkit.Bukkit;
import org.json.simple.JSONObject;

import java.util.HashMap;
import java.util.Map;

public class Telegram {
    final private Request request;

    public Telegram (String token) {
        request = new Request(String.format("https://api.telegram.org/bot%s/", token));
    }

    private void handleErrors (JSONObject response) {
        boolean isOk = Boolean.parseBoolean(response.get("ok").toString());

        if(!isOk){
            int errorCode = Integer.parseInt(response.get("error_code").toString());
            String description = response.get("description").toString();
            Bukkit.getLogger().warning(String.format("Error code: %s", errorCode));
            Bukkit.getLogger().warning(String.format("Description: %s",description));
        }
    }

    public JSONObject sendMessage(String chatId, String text) {
        Map<String, Object> data = new HashMap<>(){
            {
                put("chat_id", chatId);
                put("text", text);
            }
        };

        JSONObject response =  request.postJson("sendMessage",data);

        handleErrors(response);

        return response;
    }

    public JSONObject sendMessage(String chatId, String text, ParseMode parseMode) {
        Map<String, Object> data = new HashMap<>(){
            {
                put("chat_id", chatId);
                put("text", text);
                put("parse_mode",parseMode.toString());
            }
        };

        JSONObject response = request.postJson("sendMessage",data);

        handleErrors(response);

        return response;
    }

    public JSONObject editMessageText(String chatId, int messageId, String text) {

        Map<String, Object> data = new HashMap<>(){
            {
                put("chat_id", chatId);
                put("message_id", messageId);
                put("text", text);
            }
        };

        JSONObject response = request.postJson("editMessageText",data);

        handleErrors(response);

        return response;

    }

    public JSONObject editMessageText(String chatId, int messageId, String text, ParseMode parseMode) {

        Map<String, Object> data = new HashMap<>(){
            {
                put("chat_id", chatId);
                put("message_id", messageId);
                put("text", text);
                put("parse_mode",parseMode.toString());
            }
        };

        JSONObject response = request.postJson("editMessageText",data);

        handleErrors(response);

        return response;

    }

    public JSONObject pinChatMessage(String chatId, int messageId) {

        Map<String, Object> data = new HashMap<>(){
            {
                put("chat_id", chatId);
                put("message_id", messageId);
            }
        };

        JSONObject response = request.postJson("pinChatMessage",data);

        handleErrors(response);

        return response;

    }

}

File: src/main/kotlin/io/github/eone666/telegramnotifier/utils/telegram/Telegram.kt
package io.github.eone666.telegramnotifier.utils.telegram

import io.github.eone666.telegramnotifier.utils.Request
import org.bukkit.Bukkit
import org.json.simple.JSONObject
import kotlin.Int
import kotlin.String
import kotlin.toString

class Telegram(token: String) {

    private val request: Request = Request(String.format("https://api.telegram.org/bot%s/", token))

    private fun handleErrors(response: JSONObject?) {
        if(response != null){
            val isOk = response["ok"].toString().toBooleanStrict()
            if (!isOk) {
                val errorCode = response["error_code"].toString().toInt()
                val description = response["description"].toString()
                Bukkit.getLogger().warning(String.format("Error code: %s", errorCode))
                Bukkit.getLogger().warning(String.format("Description: %s", description))
            }
        }
    }

    private fun baseRequestOptions(chatId: String, sendSilently: Boolean) = mutableMapOf(
            "chat_id" to chatId,
            "disable_notification" to "$sendSilently"
    )

    fun sendMessage(chatId: String, sendSilently: Boolean, text: String, webPreview: Boolean, parseMode: ParseMode?): JSONObject? {
        val data = baseRequestOptions(chatId, sendSilently)
        data["text"] = text
        data["disable_web_page_preview"] = (!webPreview).toString()
        if (parseMode != null) {
            data["parse_mode"] = "$parseMode"
        }

        val response = request.postJson("sendMessage", data)
        handleErrors(response)
        return response
    }

    fun editMessageText(chatId: String, sendSilently: Boolean, messageId: Int, text: String, webPreview: Boolean, parseMode: ParseMode?): JSONObject? {
        val data = baseRequestOptions(chatId, sendSilently)
        data["message_id"] = "$messageId"
        data["text"] = text
        data["disable_web_page_preview"] = (!webPreview).toString()
        if (parseMode != null) {
            data["parse_mode"] = "$parseMode"
        }

        val response = request.postJson("editMessageText", data)
        handleErrors(response)
        return response
    }

    fun pinChatMessage(chatId: String, sendSilently: Boolean, messageId: Int): JSONObject? {
        val data = baseRequestOptions(chatId, sendSilently)
        data["message_id"] = "$messageId"

        val response = request.postJson("pinChatMessage", data)
        handleErrors(response)
        return response
    }
}